core/src/main/kotlin/live/LiveKordEntity.kt

```diff
@@ -32,20 +30,17 @@ abstract class AbstractLiveKordEntity(final override val kord: Kord, dispatcher:
 
     override val coroutineContext: CoroutineContext = dispatcher + SupervisorJob(kord.coroutineContext.job)
 
-    private val mutex = Mutex()
-
     @Suppress("EXPERIMENTAL_API_USAGE")
     final override val events: Flow<Event>
         get() = kord.events
             .takeWhile { isActive }
             .filter { filter(it) }
-            .onEach { mutex.withLock { update(it) } }
 
     protected abstract fun filter(event: Event): Boolean
     protected abstract fun update(event: Event)
 
     init {
-        events.launchIn(this)
+        events.onEach { update(it) }.launchIn(this)
     }
```

Comment:
This is indeed a problem, but I don't think your solution is the best possible.

There is a possibility that any new event listener fires *before* `update` is finished, this can be a cause of confusion since the the most obvious behaviour is that the event being listened for is already handled by the entity itself.

What we want ideally is the following:

``` 
                                          listener
                                        /
kord.events -> update -> entity.events - listener
                                        \
                                          listener
```

and kx.coroutines does offer a behaviour like that: A [Shared Flow](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-shared-flow/).

If we were to change the event flow to this:

```kotlin
final override val events: SharedFlow<Event> = kord.events.onEach { update(it) }.shareIn(this, SharingStarted.Eagerly)
```

we have:
* a flow that is eagerly subscribed
* a flow that only handles each even internally
* a flow that is cancelled when the scope dies